### PR TITLE
fix: normalize self-hosted GitLab and Gitea base URLs

### PR DESCRIPTION
**Pull request by oritwoen**
Passing a bare instance URL like `https://gitlab.example.com` to the GitLab or Gitea providers meant all API calls went to the wrong path. The fix auto-appends the right API prefix (`/api/v4` for GitLab, `/api/v1` for Gitea) when it detects a root URL, and leaves already-correct URLs alone - subpath installs included.

Tests for root URLs, subpath installs, trailing slashes, and the edge case where extra path segments follow the API prefix.

---

<a href="https://app.devin.ai/review/oritwoen/gixa/pull/5" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: src/providers/gitea.ts [hunk 1]
```diff
@@ -200,7 +201,11 @@ const PLATFORM = 'gitea';
  * @param config Provider configuration. `baseURL` defaults to `https://gitea.com/api/v1`.
  */
 export function createGiteaProvider(config: ProviderConfig): Provider {
-  const baseURL = config.baseURL ?? 'https://gitea.com/api/v1';
+  const baseURL = normalizeApiBaseURL(
+    config.baseURL,
+    'https://gitea.com/api/v1',
+    '/api/v1'
+  );
```
[the remark this answers is not in the document]
Comment (aeitwoen) [reply src/providers/gitea.ts]: Intentional. Empty baseURL is treated as fallback across providers, and GitHub/GitLab already test that behavior. Preserving '' only for Gitea would diverge from the current provider contract.
